pkg/oc/cli/cmd/rollout/latest.go

```diff
@@ -177,15 +248,35 @@ func (o RolloutLatestOptions) RunRolloutLatest() error {
 		if err != nil {
 			return err
 		}
-
-		info.Refresh(dc, true)
 	}
 
 	if o.output == "revision" {
 		fmt.Fprintf(o.out, fmt.Sprintf("%d", dc.Status.LatestVersion))
-		return nil
 	}
+	return nil
+}
 
+// RunRolloutLatest runs the latest rollouts.
+func (o RolloutLatestOptions) RunRolloutLatest() error {
+	info := o.infos[0]
+	switch t := info.Object.(type) {
+	case *deployapi.DeploymentConfig:
+		if err := o.rolloutDeploymentConfig(t); err != nil {
+			return err
+		}
+		info.Refresh(t, true)
+	case *kextensions.Deployment:
+		if err := o.rolloutDeployment(t); err != nil {
+			return err
+		}
+		info.Refresh(t, true)
+	default:
```

Comment:
@mfojtik you are missing apps  - https://kubernetes.io/docs/api-reference/v1.7/#deployment-v1beta1-apps

Reply:
oh, we are already on 1.7...

Reply:
@tnozicka reminder to sweep all the codebase and look for switches that only deal with extension.Deployments (i can think of set image for example?).

Reply:
@tnozicka talked with @kargakis, we don't have to as we use the extensions client and that will always give us extensions.Deployment... Another reminder for 3.8 or 3.9 will be to move from extensions to apps (that will also include this), but we can't do it now because of backward compatibility.

Reply:
@mfojtik you are right